README.md

```diff
@@ -1,123 +1,323 @@
- 
- * Create a [KMS wrapped key(KEK)](https://cloud.google.com/solutions/creating-cloud-dlp-de-identification-transformation-templates-pii-dataset#creating_a_key_encryption_key_kek) by creating an automatic [TEK](https://cloud.google.com/solutions/de-identification-re-identification-pii-using-cloud-dlp#token_encryption_keys) (Token Encryption Key). 
-  
- * Create DLP [inspect and re-identification template](https://cloud.google.com/solutions/creating-cloud-dlp-de-identification-transformation-templates-pii-dataset#creating_the_cloud_dlp_templates) with the KEK and crypto based transformations identified in this [section of the guide](https://cloud.google.com/solutions/de-identification-re-identification-pii-using-cloud-dlp#determining_transformation_type)
-  
- * Trigger an [automated Dataflow pipeline](https://cloud.google.com/dataflow/docs/guides/templates/provided-streaming#data-maskingtokenization-using-cloud-dlp-from-cloud-storage-to-bigquery-stream) by passing all the required [parameters](https://cloud.google.com/solutions/running-automated-dataflow-pipeline-de-identify-pii-dataset#reviewing_the_pipeline_parameters) e.g: data, configuration & dataset name.  
- 
- * Please allow 5-10 mins for the deployment to be completed.
+
+# Inspect, De-identify and Re-identify sensitive data using Cloud DLP and Dataflow
+
+> This repo contains a reference implementation of an end to end data tokenization solution. The solution is designed to migrate sensitive data to BigQuery after passing it through inspection/de-identification/re-identification Dataflow pipelines implemented using Cloud DLP. Please check out the links below for reference guides.
+
+## Table of Contents
+
+* [Architecture](#architecture)
+
+* [Concepts](#concepts)
+
+* [Tutorial](#tutorial)
+
+	* [Pre-requisites](#pre-requisites)
+    * [Compile the code](#compile-the-code)
+    * [Run the samples](#run-the-samples)
+      * [Inspection](#inspection)
+      * [De-identification](#de-identification)
+      * [Re-identification](#re-identification-from-bigquery)
+    * [Pipeline Parameters](#pipeline-parameters)
+    * [Supported File Formats](#supported-file-formats)
+    * [S3 Scanner](#s3-scanner)
+
+* [How to adapt this pipeline for your use cases](#how-to-adapt-this-pipeline-for-your-use-cases)
+
+* [Troubleshooting](#troubleshooting)
+
+* [Advanced Topics](#advanced-topics)
+
+* [Disclaimer](#disclaimer)
+
+
+## Architecture
+The solution comprises two types of pipelines (based on the DLP Transformation type). Refer [Dataflow DAG](#dataflow-dag) for pipeline job graph:
+1. Inspect and Deidentify
+2. ReIdentify
+
+### Inspection and De-identification
+![Reference Architecture](diagrams/inspect-deid-architecture.png)
+
+The pipeline can be used for CSV, TSV, Avro, JSONL files stored in either GCS or AWS S3 bucket. It uses State and Timer API for efficient batching to process the files in optimal manner.
```

Comment:
Any reason to not mention TXT format?

Reply:
Although the pipeline supports ".txt" extension, it does not offer additional benefits as compared to csv files. Basically it will work only for the csv files whose extension is changed to ".txt".

Reply:
@annadata to share his thoughts

Reply:
The pipeline supports TXT format for chat logs. We need to test it once and provide a sample along with instructions. @dup05 we should do this as a followup.